### DISCUSSION TITLE
Override type of "globalFilter"

### DISCUSSION THREAD

**Discussion by sebkasanzew** [TanStack/table#5591] [Ideas]
I want to override `globalFilter` in `GlobalFiltering.ts`, which is implemented like this:
```ts
// GlobalFiltering.ts
export interface GlobalFilterTableState {
  globalFilter: any
}
```

I tried several things like:
```ts
// react-table.d.ts in application code
declare module '@tanstack/table-core' {
  export interface GlobalFilterTableState {
    globalFilter: string
  }
}
```

but it's still typed as `any` everywhere.

The only thing that worked and made the override work, was modifying the source code in `GlobalFiltering.ts`:
```ts
// GlobalFiltering.ts
export interface GlobalFilterTableState {
  globalFilter: unknown
}
```

Is there any reason `globalFilter` needs to be typed as `any`, or is there a way to override the type without changing it to `unknown` in the source?

Here is a codesandbox with my problem: [codesandbox.io](https://codesandbox.io/p/devbox/45ydvr?file=%2Fsrc%2Fapp%2Fpage.tsx%3A92%2C24&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx1h1xpd00063b6ggl2pi0g1%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx1h1xpd00023b6gm75je37a%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx1h1xpd00043b6gjaneudce%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx1h1xpd00053b6ghs2q5us0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx1h1xpd00023b6gm75je37a%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx1h1xpd00013b6gfsp616zm%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clx1ihv9900023b6gto5nd273%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A92%252C%2522startColumn%2522%253A24%252C%2522endLineNumber%2522%253A92%252C%2522endColumn%2522%253A24%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fpage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clx1h1xpd00023b6gm75je37a%2522%252C%2522activeTabId%2522%253A%2522clx1ihv9900023b6gto5nd273%2522%257D%252C%2522clx1h1xpd00053b6ghs2q5us0%2522%253A%257B%2522id%2522%253A%2522clx1h1xpd00053b6ghs2q5us0%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clx1h61ve00fd3b6gdv5pnnve%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clx1h61ve00fd3b6gdv5pnnve%2522%257D%252C%2522clx1h1xpd00043b6gjaneudce%2522%253A%257B%2522id%2522%253A%2522clx1h1xpd00043b6gjaneudce%2522%252C%2522activeTabId%2522%253A%2522clx1h26cf005u3b6geg7sf2g1%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx1h1xpd00033b6gfy9gbk7h%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clx1h1xyz000adaf807zsfm3s%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522Development%2522%252C%2522id%2522%253A%2522clx1h26cf005u3b6geg7sf2g1%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by KevinVandy** [comment]
Just specify the actual package you have installed, not table-core

**Reply by sebkasanzew** [reply]
Thanks, I thought I tried that, but I guess not 😅

It works 👍

```ts
// react-table.d.ts in application code
declare module '@tanstack/react-table' {
  export interface GlobalFilterTableState {
    globalFilter: string
  }
}	
```

**Reply by sebkasanzew** [reply]
Ups, still had set `unknown` in `GlobalFiltering.d.ts`. So the solution above doesn't work.

I'll create a reproduction example in codesandbox later.

**Reply by sebkasanzew** [reply]
Here is a codesandbox with my problem: [codesandbox.io](https://codesandbox.io/p/devbox/45ydvr?file=%2Fsrc%2Fapp%2Fpage.tsx%3A92%2C24&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx1h1xpd00063b6ggl2pi0g1%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx1h1xpd00023b6gm75je37a%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx1h1xpd00043b6gjaneudce%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx1h1xpd00053b6ghs2q5us0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx1h1xpd00023b6gm75je37a%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx1h1xpd00013b6gfsp616zm%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clx1ihv9900023b6gto5nd273%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A92%252C%2522startColumn%2522%253A24%252C%2522endLineNumber%2522%253A92%252C%2522endColumn%2522%253A24%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fapp%252Fpage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clx1h1xpd00023b6gm75je37a%2522%252C%2522activeTabId%2522%253A%2522clx1ihv9900023b6gto5nd273%2522%257D%252C%2522clx1h1xpd00053b6ghs2q5us0%2522%253A%257B%2522id%2522%253A%2522clx1h1xpd00053b6ghs2q5us0%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clx1h61ve00fd3b6gdv5pnnve%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clx1h61ve00fd3b6gdv5pnnve%2522%257D%252C%2522clx1h1xpd00043b6gjaneudce%2522%253A%257B%2522id%2522%253A%2522clx1h1xpd00043b6gjaneudce%2522%252C%2522activeTabId%2522%253A%2522clx1h26cf005u3b6geg7sf2g1%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx1h1xpd00033b6gfy9gbk7h%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clx1h1xyz000adaf807zsfm3s%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522Development%2522%252C%2522id%2522%253A%2522clx1h26cf005u3b6geg7sf2g1%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
